app/policies/tree_policy.rb

```diff
@@ -0,0 +1,39 @@
+# frozen_string_literal: true
+
+# Policies for TreesController
+class TreePolicy < ApplicationPolicy
```

Comment:
# Discussion

This is an interesting point. I want to start a discussion here. We will have a lot more things inside of a repository (Commit, Diff, Log, Branch, Tag and many more). As far as I can see, all of them can be categorised as "repository content" and all of them should have the same set of permissions:
* allow to read if one has `read` access in the repository
* allow to create/update/delete if one has `write` access on the repository

So I suggest to clean the code base a little bit: Completely remove the TreePolicy and never add a CommitPolicy, DiffPolicy etc.. Instead, we simply call the RepositoryPolicy with `read?` and `write?`.

What do you think, @phyrog and @ebolloff?